dev/io.openliberty.org.jboss.resteasy.mprestclient/src/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java

```diff
@@ -310,10 +324,10 @@ public RestClientBuilder executorService(ExecutorService executor) {
         }
 
         if (this.executorService != null) {
-            resteasyClientBuilder.executorService(this.executorService);
+            resteasyClientBuilder.executorService(new AsyncClientExecutorService(this.executorService));
         } else {
-            this.executorService = Executors.newCachedThreadPool();
-            resteasyClientBuilder.executorService(executorService, true);
+            this.executorService = OsgiServices.getManagedExecutorService().orElseGet(Executors::newCachedThreadPool);
+            resteasyClientBuilder.executorService(new AsyncClientExecutorService(executorService), true);
```

Comment:
I think the error in the latest build is caused by these two lines. Line 330 should pass `false` if using the managed executor service and `true` if using a new cached thread pool.
```suggestion
            Optional<ExecutorService> managedExecutor = OsgiServices.getManagedExecutorService();
            this.executorService = managedExecutor.orElseGet(Executors::newCachedThreadPool);
            resteasyClientBuilder.executorService(new AsyncClientExecutorService(executorService), managedExecutor.isEmpty());
```